core/src/main/java/org/polypheny/db/adapter/DataSource.java

```diff
@@ -18,68 +18,28 @@
 
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSerializer;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import lombok.AllArgsConstructor;
+import java.util.Set;
 import lombok.Getter;
 import org.pf4j.ExtensionPoint;
 import org.polypheny.db.catalog.catalogs.AdapterCatalog;
 import org.polypheny.db.catalog.entity.LogicalAdapter.AdapterType;
-import org.polypheny.db.type.PolyType;
+import org.polypheny.db.catalog.logistic.DataModel;
 
 @Getter
 public abstract class DataSource<S extends AdapterCatalog> extends Adapter<S> implements ExtensionPoint {
 
+    private final Set<DataModel> supportedDataModels;
     private final boolean dataReadOnly;
 
 
-    protected DataSource( final long adapterId, final String uniqueName, final Map<String, String> settings, final DeployMode mode, boolean dataReadOnly, S catalog ) {
+    protected DataSource( final long adapterId, final String uniqueName, final Map<String, String> settings, final DeployMode mode, boolean dataReadOnly, S catalog, List<DataModel> supportedModels ) {
```

Comment:
`supportedModels` can be a `Set` as well.

Reply:
Thanks for the suggestion, I have changed it.